### PR TITLE
Michael -- SPS

### PR DESCRIPTION
**Pull request by MichaelCK1**
Please fill out the survey before submitting the pull request. Thanks! 

🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀

How many hours did you spend on this assignment?
about 2 hour
Please fill in one error and/or error message you received while working on this assignment.
unable to push from vscode
What part of the assignment did you spend the most time on?
second part
Comfort Level (1-5):
4
Completeness Level (1-5):
4
What did you think of this deliverable?

Is there anything in this code that you feel pleased about?
all

### CODE REVIEW

File: script.js [hunk 1]
```diff
@@ -1,4 +1,67 @@
...
+var playerName = "";
+var gameMode = "please enter player's name";
+
+var RPS = function(){
+  var randomDecimal = Math.random() * 3;
+  var randomInteger = Math.floor(randomDecimal);
+  var randomNumber = randomInteger + 1;
+  
+  return randomNumber;
 };
```
Comment (emily-zhangg) [inline script.js:15]: good work extracting certain logic intensive code that is repeated to its own helper function

File: script.js [hunk 2]
```diff
@@ -1,4 +1,67 @@
...
+  var myOutputValue = "hello! your input does not seem to be valid, please try again with either scissors, paper, or stone";
+  }
+ 
+  
+
+  if (
+    (input == SCISSORS && NPC == PAPER) ||
+    (input == PAPER && NPC == STONE) ||
+    (input == STONE && NPC == SCISSORS)
+  )  {
```
Comment (emily-zhangg) [inline script.js:48]: good use of || to combine different logic statements together

File: script.js [hunk 3]
```diff
@@ -1,4 +1,67 @@
...
+    }
+ 
+    if (
+      (input == SCISSORS && NPC == STONE) ||
+      (input == PAPER && NPC == SCISSORS) ||
+      (input == STONE && NPC == PAPER)
+    ) {
+      numOfLost = numOfLost + 1
+      myOutputValue = "You Lost! you chose "+input+" and the NPC chose "+NPC+" Your current record is " +numOfWins+" wins, " +numOfLost +" losses";
+    }
```
Comment (emily-zhangg) [inline script.js:60]: a suggestion is that maybe you would like to try to use string templating for myOutputValue, so the syntax would look like `some text here ${variables}`

**Review by yiqiingg — COMMENTED** [review]
great work michael!

**Pull request closed without merging**